pkg/apis/apps/validation/validation.go

```diff
@@ -354,6 +359,25 @@ func ValidateRollingUpdateDaemonSet(rollingUpdate *apps.RollingUpdateDaemonSet,
 	return allErrs
 }
 
+// ValidateRollingUpdateStatefulSet validates a given RollingUpdateStatefulSet.
+func ValidateRollingUpdateStatefulSet(rollingUpdate *apps.RollingUpdateStatefulSetStrategy, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs,
+		apivalidation.ValidateNonnegativeField(
+			int64(rollingUpdate.Partition),
+			fldPath.Child("partition"))...)
+	if rollingUpdate.MaxUnavailable != nil {
+		allErrs = append(allErrs, ValidatePositiveIntOrPercent(*rollingUpdate.MaxUnavailable, fldPath.Child("maxUnavailable"))...)
+		if getIntOrPercentValue(*rollingUpdate.MaxUnavailable) == 0 {
+			// MaxUnavailable cannot be 0.
```

Comment:
We do have a usecase for MaxUnavailable to be 0: can this just be a warning or be provided a way to force MaxUnavailable to 0?

Reply:
> We do have a usecase for MaxUnavailable to be 0: can this just be a warning or be provided a way to force MaxUnavailable to 0?

maxUnavailable 0 means it can never be upgraded? can you clarify your scenario ?

Reply:
@krmayankk I think that as long as the onDelete strategy is not deprecated, we're good: in practice a 0 MaxUnavailable would achieve the same result as OnDelete

Reply:
In my opinion, OnDelete should will never be deprecated. 0 is not a supported value and is not meant to replace OnDelete

Reply:
Resolved